src/Sylius/Behat/Context/Api/Admin/ManagingProductImagesContext.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\RequestBuilder;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Context\Api\Resources;
+use Sylius\Behat\Service\Converter\SectionAwareIriConverter;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\HttpFoundation\Request;
+use Webmozart\Assert\Assert;
+
+final class ManagingProductImagesContext implements Context
+{
+    public function __construct(
+        private ApiClientInterface $client,
+        private ResponseCheckerInterface $responseChecker,
+        private SharedStorageInterface $sharedStorage,
+        private \ArrayAccess $minkParameters,
+        private SectionAwareIriConverter $sectionAwareIriConverter,
+    ) {
+    }
+
+    /**
+     * @When /^I attach the "([^"]+)" image with "([^"]+)" type to (this product)$/
+     */
+    public function iAttachTheImageWithTypeToThisProduct(string $path, ?string $type, ProductInterface $product): void
+    {
+        $this->createProductImage($path, $product, $type);
+    }
+
+    /**
+     * @When /^I attach the "([^"]+)" image to (this product)$/
+     */
+    public function iAttachTheImageToThisProduct(string $path, ProductInterface $product): void
+    {
+        $this->createProductImage($path, $product);
+    }
+
+    /**
+     * @When /^I attach the "([^"]+)" image with selected ("[^"]+" variant) to (this product)$/
+     */
+    public function iAttachImageWithSelectedVariantToThisProduct(
+        string $path,
+        ProductVariantInterface $productVariant,
+        ProductInterface $product,
+    ): void {
+        $this->createProductImage($path, $product, null, [$productVariant]);
+    }
+
+    /**
+     * @When I( also) remove an image with :type type
+     */
+    public function iRemoveAnImageWithType(string $type): void
+    {
+        /** @var ProductInterface $product */
+        $product = $this->sharedStorage->get('product');
+
+        $productImage = $product->getImagesByType($type)->first();
+        Assert::notNull($productImage);
```

Comment:
`->first()` returns `false`, when nothing is found instead of `null`.
```suggestion
        Assert::notFalse($productImage);
```

Please take a look at the same at  `ManagingTaxonImagesContext.php`
```
        $taxonImage = $taxon->getImagesByType($type)->first();
        Assert::notNull($taxonImage);
```